File: Factory/SimpleFactory/TypeScript/src/Pizza/ClamPizza.ts
import Pizza from './Pizza';

class ClamPizza extends Pizza {
  constructor() {
    super();
    this.name = 'Clam Pizza';
    this.dough = 'Thin Crust';
    this.sauce = 'White Garlic Sauce';
    this.toppings.push('Clams');
    this.toppings.push('Grated Parmesan Cheese');
  }
}

export default ClamPizza;

File: Factory/SimpleFactory/Java/src/Pizza/ClamPizza.java
package eu.svetlin.design.patterns.factory.simplefactory.Pizza;

public class ClamPizza extends Pizza {
    public ClamPizza(){
        name = "Clam Pizza";
        dough = "Thin Crust";
        sauce = "White Garlic Sauce";
        toppings.add("Clams");
        toppings.add("Grated Parmesan Cheese");
    }
}
